### PR TITLE
FEC-10228 Add support in External Captions and Cast Playback of Basic Player

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    public void setHlsSource(String id, String url) {
+        Hls hlsSource =  new Hls();
+        hlsSource.id = id;
+        hlsSource.url = url;
+        hlsSource.mimetype = APPLICATION_X_MPEG_URL;
+        this.hls = Collections.singletonList(hlsSource);
+    }
+
+    public void setProgressivehSource(String id, String url) {
```
Comment (noamtamim) [inline googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: Typo

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+public class PlaybackParams {
+
+
+    public static final String PLUGINS = "plugins";
+    public static final String SOURCES = "sources";
+    public static final String APPLICATION_DASH_XML = "application/dash+xml";
+    public static final String APPLICATION_X_MPEG_URL = "application/x-mpegURL";
+    public static final String VIDEO_MP_4 = "video/mp4";
+    public static final String COM_WIDEVINE_ALPHA = "com.widevine.alpha";
```
Comment (noamtamim) [inline googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: Better call them MIMETYPE_DASH, MIMETYPE_HLS, MIMETYPE_MP4, DRMSCHEME_WIDEVINE.

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    public List<Caption> getCaptions() {
+        return captions;
+    }
+
+    public void setCaptions(List<Caption> captions) {
+        this.captions = captions;
+    }
+}
```
Comment (noamtamim) [inline googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: Do we really all of these getters and setters?

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    public JSONObject getMediaConfig() {
+        Gson gson = new Gson();
+        String sourcesJson = gson.toJson(this);
+        JSONObject playbackParams = new JSONObject();
+
+        try {
+            playbackParams.put(PLUGINS, new JSONObject());
+            playbackParams.put(SOURCES, new JSONObject(sourcesJson));
+        } catch (JSONException e ) {
+            e.printStackTrace();
```
Comment (noamtamim) [inline googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: Log?

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java (hunk 1 above)
Comment (giladna) [reply googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: done

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java (hunk 2 above)
Comment (giladna) [reply googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: done

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java (hunk 3 above)
Comment (giladna) [reply googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: no removing it

File: googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java (hunk 4 above)
Comment (giladna) [reply googlecast/src/main/java/com/kaltura/playkit/plugins/googlecast/caf/basic/PlaybackParams.java]: changed to  Log.e(TAG, e.getMessage());

**Review by noamtamim — APPROVED** [review]

**Pull request merged**